### PR TITLE
[Bug Fix]Support parameters with referenced schemas

### PR DESCRIPTION
**Pull request by aneeshafedo**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: openapi-cli/src/main/java/io/ballerina/generators/client/FunctionSignatureGenerator.java [hunk 1]
```diff
@@ -350,9 +353,20 @@ public Node getPathParameters(Parameter parameter) throws BallerinaOpenApiExcept
...
-        if (type.equals("anydata") || type.equals("[]") || type.equals("record {}")) {
-            throw new BallerinaOpenApiException(invalidPathParamType(parameter.getName().trim()));
+        String type = "";
+        Schema parameterSchema = parameter.getSchema();
+        if (parameterSchema.get$ref() != null) {
+            type = getValidName(extractReferenceType(parameterSchema.get$ref()), true);
+            Schema schema = openAPI.getComponents().getSchemas().get(type.trim());
+            if (schema instanceof ObjectSchema) {
+                throw new BallerinaOpenApiException
+                        ("Ballerina does not support to object type query parameter type.");
```
Comment (lnash94) [inline openapi-cli/src/main/java/io/ballerina/generators/client/FunctionSignatureGenerator.java:363]:
```suggestion
                        ("Ballerina does not support to object type path parameter type.");
```

File: openapi-cli/src/main/java/io/ballerina/generators/client/FunctionSignatureGenerator.java [hunk 2]
```diff
@@ -279,34 +279,37 @@ public Node getQueryParameters(Parameter parameter) throws BallerinaOpenApiExcep
...
 
-        if (parameterSchema.getType().equals("number")) {
-            if (parameterSchema.getFormat() != null) {
-                paramType = convertOpenAPITypeToBallerina(parameterSchema.getFormat().trim());
+        String paramType = "";
+        if (parameterSchema.get$ref() != null) {
+            paramType = getValidName(extractReferenceType(parameterSchema.get$ref()), true);
+            Schema schema = openAPI.getComponents().getSchemas().get(paramType.trim());
+            if (schema instanceof ObjectSchema) {
+                throw new BallerinaOpenApiException
```
Comment (lnash94) [inline openapi-cli/src/main/java/io/ballerina/generators/client/FunctionSignatureGenerator.java:288]: Shall we have this in one line?

File: openapi-cli/src/main/java/io/ballerina/generators/client/FunctionSignatureGenerator.java [hunk 3]
```diff
@@ -350,9 +353,20 @@ public Node getPathParameters(Parameter parameter) throws BallerinaOpenApiExcept
...
         IdentifierToken paramName = createIdentifierToken(escapeIdentifier(parameter.getName().trim()));
-        String type = convertOpenAPITypeToBallerina(parameter.getSchema().getType().trim());
-        if (type.equals("anydata") || type.equals("[]") || type.equals("record {}")) {
-            throw new BallerinaOpenApiException(invalidPathParamType(parameter.getName().trim()));
+        String type = "";
+        Schema parameterSchema = parameter.getSchema();
+        if (parameterSchema.get$ref() != null) {
+            type = getValidName(extractReferenceType(parameterSchema.get$ref()), true);
+            Schema schema = openAPI.getComponents().getSchemas().get(type.trim());
+            if (schema instanceof ObjectSchema) {
```
Comment (lnash94) [inline openapi-cli/src/main/java/io/ballerina/generators/client/FunctionSignatureGenerator.java:361]: Does the OpenAPI allow to have object schema in Path parameter?

File: openapi-cli/src/test/resources/generators/client/swagger/path_param_with_ref_schemas.yaml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+        "200":
+          description: Ok
+          content:
+            text/plain:
+              schema:
+                type: string
+components:
+  schemas:
+    Id:
+      type: integer
```
Comment (lnash94) [inline openapi-cli/src/test/resources/generators/client/swagger/path_param_with_ref_schemas.yaml:29]:
```suggestion
      type: integer
      
```

File: openapi-cli/src/main/java/io/ballerina/generators/client/FunctionSignatureGenerator.java (hunk 1 above)
Comment (aneeshafedo) [reply openapi-cli/src/main/java/io/ballerina/generators/client/FunctionSignatureGenerator.java]: Updated

File: openapi-cli/src/main/java/io/ballerina/generators/client/FunctionSignatureGenerator.java (hunk 3 above)
Comment (aneeshafedo) [reply openapi-cli/src/main/java/io/ballerina/generators/client/FunctionSignatureGenerator.java:360]: Yes, I have added a test case for this scenario [here](https://github.com/ballerina-platform/ballerina-openapi/blob/be005211def91caf8b1f8e46f12be42b00592c77/openapi-cli/src/test/java/io/ballerina/generators/client/PathParameterTests.java#L63)

File: openapi-cli/src/test/resources/generators/client/swagger/path_param_with_ref_schemas.yaml (hunk 4 above)
Comment (aneeshafedo) [reply openapi-cli/src/test/resources/generators/client/swagger/path_param_with_ref_schemas.yaml]: Added new line

File: openapi-cli/src/main/java/io/ballerina/generators/client/FunctionSignatureGenerator.java (hunk 2 above)
Comment (aneeshafedo) [reply openapi-cli/src/main/java/io/ballerina/generators/client/FunctionSignatureGenerator.java]: Added

File: openapi-cli/src/main/java/io/ballerina/generators/client/FunctionSignatureGenerator.java (hunk 3 above)
Comment (lnash94) [reply openapi-cli/src/main/java/io/ballerina/generators/client/FunctionSignatureGenerator.java:360]: it is fine :)

**Review by lnash94 — APPROVED** [review]

**Review by indikasampath2000 — APPROVED** [review]

**Pull request merged**